### PR TITLE
Allow custom icon specification in ScreenTitle

### PR DESCRIPTION
**Pull request by peppy**
Not all icons are available in fonts so IconUsage alone is not enough to cover all scenarios.

### CODE REVIEW

File: osu.Game/Graphics/UserInterface/ScreenTitle.cs [hunk 1]
```diff
@@ -41,6 +49,11 @@ public Color4 AccentColour
             set => pageText.Colour = value;
         }
 
+        protected virtual Drawable CreateIcon() => iconSprite = new SpriteIcon
```
Comment (smoogipoo) [inline osu.Game/Graphics/UserInterface/ScreenTitle.cs]: maybe move below ctor?

File: osu.Game/Graphics/UserInterface/ScreenTitle.cs [hunk 2]
```diff
@@ -12,26 +13,33 @@ namespace osu.Game.Graphics.UserInterface
...
+        protected const float ICON_SIZE = 25;
+
+        private SpriteIcon iconSprite;
         private readonly OsuSpriteText titleText, pageText;
-        public const float ICON_WIDTH = icon_size + icon_spacing;
-        private const float icon_size = 25, icon_spacing = 10;
+
+        private const float icon_spacing = 10;
 
         protected IconUsage Icon
```
Comment (smoogipoo) [inline osu.Game/Graphics/UserInterface/ScreenTitle.cs]: no more getter? reason?

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Graphics/UserInterface/ScreenTitle.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game/Graphics/UserInterface/ScreenTitle.cs]: Is this really where we want to be setting `iconSprite`? What if a deriver calls this method multiple times?

File: osu.Game/Graphics/UserInterface/ScreenTitle.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Graphics/UserInterface/ScreenTitle.cs]: wasn't used

File: osu.Game/Graphics/UserInterface/ScreenTitle.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Graphics/UserInterface/ScreenTitle.cs]: then they are silly

Comment (smoogipoo) [reply osu.Game/Graphics/UserInterface/ScreenTitle.cs]: what do you mean?

File: osu.Game/Graphics/UserInterface/ScreenTitle.cs (hunk 2 above)
Comment (smoogipoo) [reply osu.Game/Graphics/UserInterface/ScreenTitle.cs]: /shrug I'm not about to remove many properties in the game just because they're unused, but ok.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**